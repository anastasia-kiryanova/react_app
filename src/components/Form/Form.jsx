import React, { useEffect, useState, useRef } from 'react'
import PropTypes from 'prop-types'
import styles from './Form.module.css'
import { Button, TextField } from '@mui/material'

export const Form = ({ addMessage }) => {
  const [text, setText] = useState('')
  const inputRef = useRef(null);
  
  useEffect(() => {
      inputRef.current.focus();
   })

  const  handleText = (evt) => {
    evt.preventDefault()
    addMessage({
      text,
      author: 'User',
    })
    setText('')
  }

  const inputHandler = (evt) => {
    setText(evt.target.value)
  }

  return (
    <form className={styles.form} onSubmit={handleText}>
      <TextField name="message" variant="outlined" label="Введите сообщение" value={text} onChange={evt => inputHandler(evt)} required inputRef={inputRef} sx={{ width: '300px', mr: '30px', }} inputProps={{ 'data-testid': 'form-input' }} />
      <Button type="submit" variant="contained" color="info" data-testid="form-button">Отправить</Button>
    </form>
  )
}

Form.propTypes = {
  addMessage: PropTypes.func.isRequired
}