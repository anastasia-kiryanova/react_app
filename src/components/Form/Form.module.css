.form {
  box-sizing: border-box;
  margin: 30px 0;
  display: flex;
  justify-content: space-around;
}
