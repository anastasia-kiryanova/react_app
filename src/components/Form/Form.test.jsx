import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import { Form } from './Form';

describe ('Form', () => {
  it ('is a Component', () => {
    expect(Form).toBeInstanceOf(Function);
  });

  it ('renders Form component', () => {
    render(<Form addMessage={() => null} />);
    expect(screen.getByText('Отправить')).toBeInTheDocument();
  });

  it ('test snapshot', () => {
    const { asFragment } = render(<Form addMessage={() => null} />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe ('functionality', () => {
    beforeEach(() => {
      render(<Form />);
    });

    it ('input field is empty', () => {
      const inputEl = screen.getByTestId('form-input');
      expect(inputEl.value).toBe('');
    });

    it ('filling in the input field', () => {
      const inputEl = screen.getByTestId('form-input');
      fireEvent.change(inputEl, {target:{ value: 'abcdef123' }});
      expect(inputEl.value).toBe('abcdef123');
    });
  }); 
});