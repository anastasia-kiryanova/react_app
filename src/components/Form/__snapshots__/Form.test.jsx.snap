// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form test snapshot 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1xxymz6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="mui-2"
        id="mui-2-label"
      >
        Введите сообщение
        <span
          aria-hidden="true"
          class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           *
        </span>
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          data-testid="form-input"
          id="mui-2"
          name="message"
          required=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1z7n62"
          >
            <span>
              Введите сообщение *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-kjirbw-MuiButtonBase-root-MuiButton-root"
      data-testid="form-button"
      tabindex="0"
      type="submit"
    >
      Отправить
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
</DocumentFragment>
`;
